### PR TITLE
examples: NonArduino: Tock: Initial commit

### PR DESCRIPTION
**Pull request by alistair23**
Add support for running RadioLib on Tock.

Tock is an embedded operating system designed for running multiple concurrent, mutually distrustful applications on Cortex-M and RISC-V based embedded platforms (https://github.com/tock/tock).

This PR uses libtock-c (https://github.com/tock/libtock-c) to add support to running RadioLib as a Tock userspace application.

This has been tested on the SparkFun LoRa Thing Plus - expLoRaBLE board (https://github.com/tock/tock/tree/master/boards/apollo3/lora_things_plus) but will work on any LoRa compatible Tock board (currently only the expLoRaBLE board).

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "examples/NonArduino/Tock/libtock-c"]
```
Comment (jgromes) [inline .gitmodules:1]: This submodule will not be available if someone installs RadioLib as an Arduino Library (via Arduino library manager). Probably not an issue, but I wanted to mention it.

File: examples/NonArduino/Tock/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# the following is just an example, yours will likely be different
+add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/../../../../RadioLib" "${CMAKE_CURRENT_BINARY_DIR}/RadioLib")
+
+# add the executable
+add_executable(${PROJECT_NAME} main.cpp)
+
+# link with RadioLib and libtock-c
+target_link_libraries(${PROJECT_NAME} PUBLIC
+    RadioLib
+    gcc
```
Comment (jgromes) [inline examples/NonArduino/Tock/CMakeLists.txt:24]: Is linking to gcc really needed? Just curious.

File: examples/NonArduino/Tock/libtockHal.h [hunk 3]
```diff
@@ -0,0 +1,170 @@
+#ifndef ESP_HAL_H
+#define ESP_HAL_H
```
Comment (jgromes) [inline examples/NonArduino/Tock/libtockHal.h:2]: Probably leftover from ESP-IDF

File: examples/NonArduino/Tock/main.cpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  // pinout corresponds to the SparkFun LoRa Thing Plus - expLoRaBLE
+  // NSS pin:   0
+  // DIO1 pin:  2
+  // NRST pin:  4
+  // BUSY pin:  1
+  Module* tock_module = new Module(hal, RADIO_NSS, RADIO_DIO_1, RADIO_RESET, RADIO_BUSY);
+  SX1262* radio = new SX1262(tock_module);
+
+  // Module *mod = radio.getMod();
+  // mod->SPItransfer = SPItransfer;
```
Comment (jgromes) [inline examples/NonArduino/Tock/main.cpp:29]: Probably a forgotten commented-out code.

File: examples/NonArduino/Tock/tock.cmake [hunk 5]
```diff
@@ -0,0 +1,30 @@
+## License:  Revised BSD License, see LICENSE.TXT file included in the project
```
Comment (jgromes) [inline examples/NonArduino/Tock/tock.cmake:1]: RadioLib license is MIT, this is most likely referencing a different file. If so, please include this license in the example.

File: examples/NonArduino/Tock/toolchain-arm-none-eabi.cmake [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+##  _____) ) ____| | | || |_| ____( (___| | | |
+## (______/|_____)_|_|_| \__)_____)\____)_| |_|
+## (C)2013-2017 Semtech
+##  ___ _____ _   ___ _  _____ ___  ___  ___ ___
+## / __|_   _/_\ / __| |/ / __/ _ \| _ \/ __| __|
+## \__ \ | |/ _ \ (__| ' <| _| (_) |   / (__| _|
+## |___/ |_/_/ \_\___|_|\_\_| \___/|_|_\\___|___|
+## embedded.connectivity.solutions.==============
+##
+## License:  Revised BSD License, see LICENSE.TXT file included in the project
```
Comment (jgromes) [inline examples/NonArduino/Tock/toolchain-arm-none-eabi.cmake:15]: Same comment as before

**Review by jgromes — CHANGES_REQUESTED** [review]
Thank you, I left some comments. The only major is the point about license.

File: examples/NonArduino/Tock/CMakeLists.txt (hunk 2 above)
Comment (alistair23) [reply examples/NonArduino/Tock/CMakeLists.txt:48]: Yeah it is. libtock-c requires some of the helper functions provided

**Comment by alistair23** [comment]
I have updated the license information in each file to be explicit and clear

**Comment by jgromes** [comment]
All good now, thanks for the contribution!

**Pull request merged**